aim2dat/strct/strct_manipulation.py

```diff
@@ -114,15 +111,58 @@ def substitute_elements(
             new_structure["elements"]
         )
         if structure["cell"] is not None:
-            new_structure["cell"] = np.array(structure["cell"])
-            for dir_idx in range(3):
-                new_structure["cell"][dir_idx] *= scaling_factor
+            new_structure["cell"] = [
+                [value * scaling_factor if i == dir_idx else value for i, value in enumerate(row)]
+                for dir_idx, row in enumerate(structure["cell"])
+            ]
         return _add_label_suffix(new_structure, "_subst-" + "-".join(str_el_pairs), change_label)
 
 
-def scale_unit_cell(structure: Structure, scaling_factor: float, change_label: bool) -> Structure:
-    """Scale unit cell of the structure."""
-    if scaling_factor != 1.0 and structure["cell"] is not None:
-        new_structure = structure.to_dict(cartesian=False)
-        new_structure["cell"] = np.array(structure["cell"]) * scaling_factor
-        return _add_label_suffix(new_structure, f"_scaled-{scaling_factor}", change_label)
+def scale_unit_cell(
+    structure,
+    scaling_factors: Union[float, List[float]] = None,
+    pressure: float = None,
+    bulk_modulus: float = None,
+    change_label: bool = True,
+) -> "Structure":
+    """Scale the unit cell of a structure."""
+    from aim2dat.strct import Structure
+
+    def get_scaling_matrix(scaling_factors):
+        """Construct a 3x3 scaling matrix."""
+        if isinstance(scaling_factors, (float, int)):
+            return [[scaling_factors if i == j else 0 for j in range(3)] for i in range(3)]
+        if isinstance(scaling_factors, list):
+            if len(scaling_factors) == 3 and all(
+                isinstance(val, (float, int)) for val in scaling_factors
+            ):
+                return [[scaling_factors[i] if i == j else 0 for j in range(3)] for i in range(3)]
+            if len(scaling_factors) == 3 and all(
+                isinstance(row, list) and len(row) == 3 for row in scaling_factors
+            ):
+                return scaling_factors
+        raise ValueError(
+            "Scaling factors must be a float, a list of 3 values, or a 3x3 nested list."
+        )
```

Comment:
Yes, this is what I had in mind, i.e., unifying the behavior for lists and arrays, while still supporting both of them. Thanks @hdsassnick. Just one additional comment. Your current suggestion would fail in case the scaling factors are a row or column vector, i.e. having a shape of `(3, 1)` or `(1, 3)`. Hence, I'd suggest to change line `141` of your suggestion to `elif scaling_factors.size == 3`. `size` simply counts the number of elements.

Reply:
Thanks @t-reents for the input. I am just wondering whether there is an actual difference between row and column vectors? The following code snippet outputs True..
```
import numpy as np

a = np.array([1, 2, 3])
a.shape == a.T.shape
```

In any case, I adjusted the suggestion above with the suggested size attribute.

Reply:
Hi, indeed, in your test case there is no difference, since `a` is a flat array, i.e. `a.shape --> (3,)`. However, if you have a row vector (which is in principle a 2D array, but I still think that you might want to support it), e.g., `a = np.array([[1, 2, 3]])`, you have a shape of `a.shape --> (1, 3)` and `a.T.shape --> (3, 1)`. Hence, the comparison of the shapes would return `False`

Reply:
Thanks for the clarification. I just double-checked and the current (edited) version should support the 2D row/column array case as well.